### PR TITLE
Add initial popularity to websites when importing (bug 1161858)

### PR DESCRIPTION
**Pull request by diox**
https://bugzilla.mozilla.org/show_bug.cgi?id=1161858

The value is "stolen" from apps, to provide a healthy initial mix of apps and websites in new and popular pages at launch.

### CODE REVIEW

File: mkt/websites/management/commands/import_websites_from_csv.py [hunk 1]
```diff
@@ -181,6 +193,7 @@ def set_url(self, instance, row):
                 u'Website %s has no URL ?!' % row['Unique Moz ID'])
 
     def set_icon(self, instance, row):
+        return
```
Comment (robhudson) [inline mkt/websites/management/commands/import_websites_from_csv.py]: ?

Comment (diox) [inline mkt/websites/management/commands/import_websites_from_csv.py]: oops, debugging I meant to remove.

File: mkt/websites/management/commands/import_websites_from_csv.py [hunk 2]
```diff
@@ -202,8 +215,65 @@ def parse(self, filename):
...
+            print u'Setting regional popularity for %d site(s) in %s' % (
+                websites_len, unicode(REGIONS_CHOICES_ID_DICT[region].name))
+            # Sort sites by rank in that region.
+            websites = sorted(self.ranking_per_region[region],
+                              key=itemgetter(1), reverse=True)
+            # Take the same number of popularity values for apps in that
+            # region.
+            apps_popularity = (Installs.objects.filter(region=region)
+                                       .values_list('value', flat=True)
+                                       .order_by('-value')[:websites_len])
```
Comment (robhudson) [inline mkt/websites/management/commands/import_websites_from_csv.py]: `Installs` only counts installs for users who are logged in. This won't be the same as the popularity we store in the popularity tables which also includes anonymous installs.

Comment (diox) [inline mkt/websites/management/commands/import_websites_from_csv.py]: `Installed` is for the users who are logged in, `Installs` is the popularity table for apps. It has this FK:
`addon = models.ForeignKey(Webapp, related_name='popularity')`

Comment (robhudson) [inline mkt/websites/management/commands/import_websites_from_csv.py]: Ah, good. Ignore me then. :)

File: mkt/websites/management/commands/import_websites_from_csv.py [hunk 3]
```diff
@@ -257,6 +332,16 @@ def handle(self, *args, **kwargs):
...
         self.overwrite = kwargs.get('overwrite', False)
         self.limit = kwargs.get('limit', None)
+        self.set_popularity = kwargs.get('set_popularity', False)
+
+        if self.set_popularity:
+            if self.limit:
+                raise CommandError(
+                    'Can not use --set_popularity with --limit, the full data '
+                    'set is needed to set popularity, aborting.')
+                return
```
Comment (robhudson) [inline mkt/websites/management/commands/import_websites_from_csv.py]: if you `raise` do you also need the `return`?

File: mkt/websites/management/commands/import_websites_from_csv.py [hunk 4]
```diff
@@ -202,8 +214,65 @@ def parse(self, filename):
...
+        print 'Setting global popularity values...'
+        GLOBAL_REGION = 0
+        for pk in self.websites:
+            values = list(WebsitePopularity.objects
+                                           .filter(website=pk)
+                                           .exclude(region=GLOBAL_REGION)
+                                           .values_list('value', flat=True))
+            popularity, created = WebsitePopularity.objects.get_or_create(
+                website_id=pk, region=GLOBAL_REGION)
+            popularity.update(value=sum(values))
```
Comment (robhudson) [inline mkt/websites/management/commands/import_websites_from_csv.py]: I had to read this a few times to understand everything, sorry for the review delay.

File: mkt/websites/management/commands/import_websites_from_csv.py [hunk 5]
```diff
@@ -202,8 +214,65 @@ def parse(self, filename):
...
+        last_updated_dates = list(
+            Webapp.objects
+                  .exclude(last_updated=None)
+                  .values_list('last_updated', flat=True)
+                  .order_by('-last_updated')[:desired_len])
+        if len(last_updated_dates) < desired_len:
+            raise CommandError('Not enough apps with a last_updated set in the'
+                               ' database to continue!')
+            return
+        random.shuffle(last_updated_dates)
```
Comment (robhudson) [inline mkt/websites/management/commands/import_websites_from_csv.py]: Let's all bow our heads to RNGesus.

**Comment by robhudson** [comment]
r+wc

**Pull request merged**